lib/Controller/PageController.php

```diff
@@ -68,12 +68,12 @@ public function index(): TemplateResponse {
 		$this->eventDispatcher->dispatch(LoadViewer::class, new LoadViewer());
 
 		$this->initialStateService->provideInitialState($this->appName, 'mimes', Application::MIMES);
+		$this->initialStateService->provideInitialState($this->appName, 'maps', Application::MAPS);
```

Comment:
`Application::MAPS` doesn't exists,
you need to fetch the proper data for it.
You can check if an app is enabled by using this
`$this->config->getAppValue('maps', 'enabled', 'no') === 'yes')`
Make sure you inject the config dependency in this file's constructor as well

Reply:
Sorry, no idea what to do here. 
I don't know where to add "$this->config->getAppValue('maps', 'enabled', 'no') === 'yes')" 
and what "Make sure you inject the config dependency in this file's constructor as well" could mean...

Reply:
Are you familiar with our nextcloud app ecosystem? :)
We updated our tutorial recently here: https://docs.nextcloud.com/server/stable/developer_manual/app/tutorial.html

the provideInitialState requires 3 argments, the name of the app, the unique key of the data set and the data you want to pass.
Here we want to pass if the maps is enabled or not (boolean).
`Application::MAPS` doesn't exists. Look at the Application.php file.
https://github.com/nextcloud/photos/blob/master/lib/AppInfo/Application.php

Reply:
So you need to pass the data we want, meaning `$this->config->getAppValue('maps', 'enabled', 'no') === 'yes')` that will return true/false if maps is enabled.

BUT $this->config is not injected here
https://github.com/nextcloud/photos/blob/c73e919829fd43d0f133609a8457790d447baa13/lib/Controller/PageController.php#L47-L55
You need to import [the IConfig interface](https://docs.nextcloud.com/server/stable/developer_manual/api/OCP/IConfig.html) the same way we inject the IRequest or IEventDispatcher ;)

Then you will be able to pass `$this->config->getAppValue('maps', 'enabled', 'no') === 'yes')` as the maps initial state

Reply:
ANd the nextcloud php ecosystem will do the rest and give you access to `$config`, which you can then assign to `$this->config` and use it :)

Reply:
Well done :)